### PR TITLE
Adds a length method to calculate a time-window's length.

### PR DESCRIPTION
**Pull request by joeldudleyr3**

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/contracts/TimeWindow.kt [hunk 1]
```diff
@@ -64,6 +64,17 @@ abstract class TimeWindow {
      */
     abstract val midpoint: Instant?
 
+    /**
+     * Returns the duration between [fromTime] and [untilTime] if both are non-null. Otherwise returns null.
+     */
```
Comment (adagys) [inline core/src/main/kotlin/net/corda/core/contracts/TimeWindow.kt]: can use Duration.INDEFINITE instead of null

Comment (joeldudleyr3) [reply core/src/main/kotlin/net/corda/core/contracts/TimeWindow.kt]: I can't find it - where is it? https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html

Comment (adagys) [reply core/src/main/kotlin/net/corda/core/contracts/TimeWindow.kt]: ah, yes it's not in the java.time one

File: core/src/test/kotlin/net/corda/core/contracts/TimeWindowTest.kt [hunk 2]
```diff
@@ -1,9 +1,12 @@
 package net.corda.core.contracts
 
+import net.corda.core.internal.div
+import net.corda.core.internal.times
```
Comment (kchalkias) [inline core/src/test/kotlin/net/corda/core/contracts/TimeWindowTest.kt]: do we use those two?

Comment (joeldudleyr3) [reply core/src/test/kotlin/net/corda/core/contracts/TimeWindowTest.kt]: Yes, for dividing/multiplying doubles in the new tests.

Comment (kchalkias) [reply core/src/test/kotlin/net/corda/core/contracts/TimeWindowTest.kt]: just noticed

**Comment by kchalkias** [comment]
Could you also add changelog notes?

**Comment by joeldudleyr3** [comment]
Added a change-log entry.

**Pull request merged**